File: api-asn/csharp/src/api/asn/pkcs1pkcs8/ERSAPrivateKey.cs
using Com.Objsys.Asn1.Runtime;
using tr.gov.tubitak.uekae.esya.asn.pkcs1pkcs8;

namespace tr.gov.tubitak.uekae.esya.api.asn.pkcs1pkcs8
{
    public class ERSAPrivateKey : BaseASNWrapper<RSAPrivateKey>
    {
        public ERSAPrivateKey(RSAPrivateKey aRsaPrivateKey)
            : base(aRsaPrivateKey)
        {
        }

        public ERSAPrivateKey(EPrivateKeyInfo aPrivateKeyInfo)
            : this(aPrivateKeyInfo.getPrivateKey())
        {
        }

        public ERSAPrivateKey(byte[] aBytes)
            : base(aBytes, new RSAPrivateKey())
        {
        }

        public BigInteger getPrivateExponent()
        {
            return mObject.privateExponent.mValue;
        }

        public BigInteger getModulus()
        {
            return mObject.modulus.mValue;
        }

        public BigInteger getPublicExponent()
        {
            return mObject.publicExponent.mValue;
        }

        public BigInteger getPrimeP()
        {
            return mObject.prime1.mValue;
        }

        public BigInteger getPrimeQ()
        {
            return mObject.prime2.mValue;
        }

        public BigInteger getPrimeExponentP()
        {
            return mObject.exponent1.mValue;
        }

        public BigInteger getPrimeExponentQ()
        {
            return mObject.exponent2.mValue;
        }

        public BigInteger getCrtCoefficient()
        {
            return mObject.coefficient.mValue;
        }

    }
}

File: api-asn/src/tr/gov/tubitak/uekae/esya/api/asn/pkcs1pkcs8/ERSAPrivateKey.java
package tr.gov.tubitak.uekae.esya.api.asn.pkcs1pkcs8;

import java.math.BigInteger;


import tr.gov.tubitak.uekae.esya.api.asn.BaseASNWrapper;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.asn.pkcs1pkcs8.RSAPrivateKey;

public class ERSAPrivateKey extends BaseASNWrapper<RSAPrivateKey>{
	
	public ERSAPrivateKey(byte[] aBytes) throws ESYAException {
        super(aBytes, new RSAPrivateKey());
    }

	public BigInteger getPrivateExponent() {
		return mObject.privateExponent.value;
	}

	public BigInteger getModulus() {
		return mObject.modulus.value;
	}

	public BigInteger getPublicExponent() {
		return mObject.publicExponent.value;
	}

	public BigInteger getPrimeP() {
		return mObject.prime1.value;
	}

	public BigInteger getPrimeQ() {
		return mObject.prime2.value;
	}

	public BigInteger getPrimeExponentP() {
		return mObject.exponent1.value;
	}

	public BigInteger getPrimeExponentQ() {
		return mObject.exponent2.value;
	}

	public BigInteger getCrtCoefficient() {
		return mObject.coefficient.value;
	}
}
